File: baselibrary/src/main/java/com/aslan/baselibrary/base/VBBaseDialog.java
package com.aslan.baselibrary.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;
import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.lang.reflect.ParameterizedType;

/**
 * 基础类
 *
 * @author Aslan
 * @date 2018/4/11
 */
public abstract class VBBaseDialog<VB extends ViewBinding> extends BaseDialogFragment {

  protected VB mViewBinding;

  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    ParameterizedType type = (ParameterizedType) getClass().getGenericSuperclass();
    Class cls = (Class) type.getActualTypeArguments()[0];
    try {
      Method inflate = cls
          .getDeclaredMethod("inflate", LayoutInflater.class, ViewGroup.class, boolean.class);
      mViewBinding = (VB) inflate.invoke(null, inflater, container, false);
    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
      e.printStackTrace();
    }
    return mViewBinding.getRoot();
  }

  @Override
  public final void iniView(@NonNull View view) {
    iniView();
  }

  public abstract void iniView();

  @Override
  public void onDestroy() {
    super.onDestroy();
    mViewBinding = null;
  }
}

File: baselibrary/src/main/java/com/aslan/baselibrary/base/VBBaseDialog.kt
package com.aslan.baselibrary.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.aslan.baselibrary.utils.InflateFragment

/**
 * 基础类
 *
 * @author Aslan
 * @date 2018/4/11
 */
abstract class VBBaseDialog<VB : ViewBinding>(private val inflate: InflateFragment<VB>) :
    BaseDialogFragment() {
    private var _binding: VB? = null
    protected val mViewBinding get() = _binding!!

    override fun getLayoutId() = 0

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = inflate.invoke(inflater, container, false)
        return mViewBinding.root
    }

    override fun iniView(view: View) {
        iniView()
    }

    abstract fun iniView()

    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }
}